modules/fs2/src/main/scala/LangoustineApp.scala

```diff
@@ -0,0 +1,135 @@
+package langoustine
+package lsp
+package fs2
+
+import cats.effect.IOApp
+import cats.effect.ExitCode
+import cats.effect.IO
+import jsonrpclib.fs2.*
+import _root_.fs2 as FS2
+import cats.effect.std.Dispatcher
+import scala.concurrent.Future
+import langoustine.lsp.requests.LSPNotification
+import langoustine.lsp.requests.LSPRequest
+import jsonrpclib.Channel
+import jsonrpclib.Endpoint.NotificationEndpoint
+import jsonrpclib.Endpoint.RequestResponseEndpoint
+import jsonrpclib.Endpoint
+import cats.effect.kernel.Resource
+import cats.syntax.all.*
+
+trait LangoustineApp extends IOApp with LangoustineApp.Config:
+  def server(args: List[String]): IO[LSPBuilder[IO]]
+
+  override def run(args: List[String]): IO[ExitCode] =
+    FS2.Stream
+      .eval(server(args))
+      .flatMap { builder =>
+        LangoustineApp.create(lspBufferSize, builder, in, out)
+      }
+      .compile
+      .drain
+      .guarantee(IO.consoleForIO.errorln("Terminating server"))
+      .as(ExitCode.Success)
+end LangoustineApp
+
+def bind(builder: LSPBuilder[Future], to: Channel[IO]) =
```

Comment:
I will not be taking any questions.

TODO: make private, move from one huge file.